env0/resource_module.go

```diff
@@ -0,0 +1,199 @@
+package env0
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceModule() *schema.Resource {
+	vcsExcatlyOneOf := []string{
+		"token_id",
+		"github_installation_id",
+		"bitbucket_client_key",
+	}
+
+	return &schema.Resource{
+		CreateContext: resourceModuleCreate,
+		ReadContext:   resourceModuleRead,
+		UpdateContext: resourceModuleUpdate,
+		DeleteContext: resourceModuleDelete,
+
+		Importer: &schema.ResourceImporter{StateContext: resourceModuleImport},
+
+		Schema: map[string]*schema.Schema{
+			"module_name": {
+				Type:             schema.TypeString,
+				Description:      "name of the module",
```

Comment:
It's quite common in TF registry docs to also specify any attribute constraint to make any user lives easier. Please also note the name constraint to the attribute description

Reply:
https://github.com/env0/terraform-provider-env0/pull/279/commits/a59b4bbbb4c4f0f861546b4848b2c55e5040adbb